core/app/models/spree/role.rb

```diff
@@ -3,12 +3,26 @@
 module Spree
   class Role < Spree::Base
     has_many :role_users, class_name: "Spree::RoleUser", dependent: :destroy
+    has_many :role_permissions, dependent: :destroy
+    has_many :permission_sets, through: :role_permissions
     has_many :users, through: :role_users
 
+    scope :non_base_roles, -> { where.not(name: ['admin', 'default']) }
```

Comment:
What do you think about storing the reserved role names like 'admin' and 'default' in a constant? 🤔 like `RESERVED_ROLES` or `STATIC_ROLES` something like that.